### DISCUSSION TITLE
Exporting alpha vectors.

### DISCUSSION THREAD

**Discussion by aidmandorky** [JuliaPOMDP/POMDPs.jl#443] [Q&A]
Apologies for the noob question, but I am trying to wrap my head around the POMDP ecosystem in Julia. I had a couple of questions:

- Is it possible to export any problem I define within Julia as a POMDPX file, or a [Cassandra POMDP file](http://pomdp.org/code/pomdp-file-spec.html)?

- Is it possible to export the alpha vectors as a plain text file like the `.alpha` files output by the Cassandra `pomdp-solve` solver after the solution has been found? 

Thanks in advance!

**Comment by zsunberg** [comment]
> Is it possible to export any problem I define within Julia as a POMDPX file

Yes, via https://github.com/JuliaPOMDP/POMDPXFiles.jl though I am not sure if every feature of the POMDPX format is supported. This is used routinely with SARSOP.jl, so it should work in most cases.

> or a [Cassandra POMDP file](http://pomdp.org/code/pomdp-file-spec.html)?

There is a package that attempts to do this https://github.com/JuliaPOMDP/POMDPFiles.jl but it is much less often used than the POMDPXFiles package, so I would not trust it very much

> Is it possible to export the alpha vectors as a plain text file like the .alpha files output by the Cassandra pomdp-solve solver after the solution has been found?

I am not sure if this code already exists anywhere, but alpha vector policies are usually represented with an [`AlphaVectorPolicy`](https://juliapomdp.github.io/POMDPs.jl/latest/POMDPTools/policies/#Alpha-Vector-Policy), and it should be fairly easy to iterate through the vectors and write them to the file with proper bookkeeping about the order of states and actions, etc. Let us know if you need more help with that.
